shanoir-ng-import/src/main/java/org/shanoir/ng/importer/ImporterApiController.java

```diff
@@ -781,4 +801,141 @@ public ResponseEntity<ByteArrayResource> getDicomImage(@ApiParam(value = "path",
+			File[] examinationsFolders = subjectFolder.listFiles();
+
+			String subjectName = studyName + "_" + subjectFolder.getName();
+			Subject subject = null;
+
+
+			// STEP 4: Iterate over examination folders
+			for (File examFolder : examinationsFolders) {
+				job = null;
+				// Check of it's a folder
+				if (!examFolder.isDirectory()) {
+					throw new RestServiceException(new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(),
+							"The main subject folder should only contain sub-folders and not single/data files.", null));
+				}
+				File zippedExamFolder = new File (examFolder.getAbsolutePath() + ".zip");
+				zippedExamFolder.createNewFile();
+				Utils.zip(examFolder.getAbsolutePath(), zippedExamFolder.getAbsolutePath());
+				MockMultipartFile mockedFile = new MockMultipartFile(examFolder.getName(), zippedExamFolder.getName(), APPLICATION_ZIP, new FileInputStream(zippedExamFolder));
+
+				// STEP 4.0 "Fake" upload file to get create the dicomdir and temporary folder
+				job = this.uploadDicomZipFile(mockedFile).getBody();
+
+				// Create subject only once.
+				if (subject == null) {
+					Patient pat = job.getPatients().get(0);
+					// Create subject
+					subject = new Subject();
+					subject.setName(subjectName);
+					subject.setBirthDate(pat.getPatientBirthDate());
+					subject.setSex(pat.getPatientSex());
+					// LIVING_HUMAN_BEING by default
+					subject.setImagedObjectCategory(1);
+					// Here the ID is used to carry the study ID
+					subject.setId(studyId);
+
+					LOG.debug("We found a subject " + subjectName);
+
+					// Create subject
+					Long subjectId = (Long) rabbitTemplate.convertSendAndReceive(RabbitMQConfiguration.SUBJECTS_QUEUE, objectMapper.writeValueAsString(subject));
+					if (subjectId == null) {
+						throw new RestServiceException(new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), "Subject oculd not be created, please check data", null));
+					}
+					subject.setId(subjectId);
+					// Wait a second for the subject to be created dataset side ?
+					Thread.sleep(1000);
+				}
+
+				// STEP 4.1 Get informations about center / study card
+				// TODO: Do this only if modality = MR !!
+				// Get equipment id
+				if (job.getPatients().get(0).getStudies().get(0).getSeries().get(0).getEquipment().getDeviceSerialNumber() != null) {
+					Long equipmentId = (Long) this.rabbitTemplate.convertSendAndReceive(RabbitMQConfiguration.EQUIPMENT_FROM_CODE_QUEUE, job.getPatients().get(0).getStudies().get(0).getSeries().get(0).getEquipment().getDeviceSerialNumber());
+					if (equipmentId != null) {
+						Properties props = new Properties();
+						props.setProperty("EQUIPMENT_ID_PROPERTY", "" + equipmentId);
+						props.setProperty("STUDY_ID_PROPERTY", "" + studyId);
+						props.setProperty("STUDYCARD_ID_PROPERTY", "" + studyCardId);
+
+						Long newStudyCardId = (Long) this.rabbitTemplate.convertSendAndReceive(RabbitMQConfiguration.IMPORT_STUDY_CARD_QUEUE, props);
```

Comment:
sorry, but please sync call as well

Reply:
When there is a value in return, the call is synchronous